src/wp-includes/class-wp-query.php

```diff
@@ -4039,6 +4039,10 @@ public function is_page( $page = '' ) {
 
 		$page_obj = $this->get_queried_object();
 
+		if ( ! isset( $page_obj->ID ) && ! isset( $page_obj->post_title ) && ! isset( $page_obj->post_name ) ) {
```

Comment:
only checking if $page_obj is empty will be enough. if $page_object value is null, below codes doesn't need to execute, kindly change this to:
```
if ( empty( $page_obj->ID ) ) {
    return false;
}
```